test/unit/subnet/ipv4_test.rb

```diff
@@ -127,6 +133,13 @@ def create_a_domain_with_the_subnet
     assert s.save
   end
 
+  test "should not be able to save ranges if one of them is invalid" do
+    s=subnets(:one)
+    s.from = "2.3.4.abc"
+    s.to   = "2.3.4.17"
+    assert !s.save
```

Comment:
Or even `should_not allow_value('2.3.4.abc').for(:from)` , shoulda-matchers syntax.